### PR TITLE
Add makefile and update CI to use make targets

### PR DESCRIPTION
**Pull request by nolancon**
If you choose to merge this PR, then the README should probably be updated to show new make target too 😎

Also, I know there was no issue for this, but I think it makes the user's life easier. All in the interest of Hacktoberfest 😄

### CODE REVIEW

**Comment by Gituser143** [comment]
This is a great idea! Although I'm not very sure of how the modified GitHub workflows would work. I'll test around a bit before merging this.

Edit: Workflows work like a charm!

**Comment by Gituser143** [comment]
@nolancon could you please add "make" subsection in the readme for building? Apart from that everything looks perfect!

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+##@ Docker
+docker-build: ## Build docker image.
+	docker build -t cryptgo .
+
+docker-pull: ## Pull docker image.
+	docker pull bhargavsnv/cryptgo:$(VERSION)
+
+docker-run: ## Run docker image.
+	docker run -u $(id -u):$(id -g) -v "$(HOME):/home/appuser/" --rm -it bhargavsnv/cryptgo:$(VERSION)
```
Comment (Gituser143) [inline Makefile]: Could this be extended to take arguments too? `cryptgo portfolio` or `cryptgo --help` and such commands cannot be run using this currently.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,40 @@
+VERSION ?= latest
+
+
+##@ General
+
+help:  ## Display this help.
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_0-9-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
```
Comment (Gituser143) [inline Makefile:8]: This is absolutely brilliant! Tedious, but I ❤️ it nonetheless :P

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+build: ## Build binary.
+	go build -v ./... 
+
+all: deps test ## Install deps, test and build binary.
+	make build
+
+##@ Docker
+docker-build: ## Build docker image.
+	docker build -t cryptgo .
```
Comment (Gituser143) [inline Makefile]: This is on me, but could this be changed to `bhargavsnv/cryptgo` instead of just `cryptgo`?

**Review by Gituser143 — CHANGES_REQUESTED** [review]

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+test: fmt vet license ## Run unit tests.
+	go test -v ./...
+
+##@ Build
+deps: ## Install dependencies
+	go mod download
+
+build: ## Build binary.
+	go build -v ./...
```
Comment (Gituser143) [inline Makefile]: @nolancon could you replace this with `go build cryptgo.go` as this creates and places a binary while `go build -v ./...` does not.

**Review by Gituser143 — CHANGES_REQUESTED** [review]

**Comment by Gituser143** [comment]
Also, could you please avoid force pushing? Multiple commits are fine, I'll be squashing them before merge anyway.

**Review by Gituser143 — APPROVED** [review]
Everything looks good! If there's nothing else you had in mind to add, I'll merge this?

**Comment by nolancon** [comment]
> Everything looks good! If there's nothing else you had in mind to add, I'll merge this?

Nothing else. You can go ahead and merge 👍

**Comment by Gituser143** [comment]
@nolancon Thank you for your contribution!

**Pull request merged**